risc0/zkvm/src/prove/mod.rs

```diff
@@ -415,7 +421,7 @@ impl<'a> Prover<'a> {
 
         // Attach the full version of the output journal & construct receipt object
         let receipt = Receipt {
-            journal: self.inner.journal.clone(),
+            journal: take(&mut self.inner.journal.buf.borrow_mut()),
```

Comment:
Using `RefCell::take` has the nice property that it will panic if there is an outstanding borrow.
```suggestion
            journal: self.inner.journal.buf.take(),
```